module-db/Interface/NotificationsRecord.cpp

```diff
@@ -170,7 +170,7 @@ std::unique_ptr<db::query::notifications::GetResult> NotificationsRecordInterfac
 std::unique_ptr<db::query::notifications::IncrementResult> NotificationsRecordInterface::runQueryImpl(
     const db::query::notifications::Increment *query)
 {
-    auto ret = processIncrement(query->getKey(), query->getNumber(), 1);
+    auto ret = processIncrement(query->getKey(), query->getNumber(), query->getCount());
```

Comment:
so what will happen here when we get the negative count? :)

Reply:
Is it possible that query returns negative count? I haven't checked all getCount methods, however it seems that all of those return unsigned value. Nevertheless, we could check for it, just in case.

Reply:
In case of a negative count, the value of notifications in the record will be decremented.